src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -572,19 +575,93 @@ namespace OpenMS
+      std::vector<IdentificationData::InputFileRef> file_refs;
+      file_refs.emplace_back(file_ref);
+
+      // add previous DataProcessingStep(s) from FeatureMap
+      auto data_processing = fmap.getDataProcessing();
+      for (const auto& it : data_processing)
+      {
+        // software
+        IdentificationData::ProcessingSoftware sw(it.getSoftware().getName(), it.getSoftware().getVersion());
+        // transfer previous metadata
+        sw.addMetaValues(it);
+        IdentificationDataInternal::ProcessingSoftwareRef sw_ref = id.registerProcessingSoftware(sw);
+        // ProcessingStep: software, input_file_refs, data_time, actions
+        IdentificationData::ProcessingStep step(sw_ref, file_refs, it.getCompletionTime(), it.getProcessingActions());
+        step_ref = id.registerProcessingStep(step);
+        id.setCurrentProcessingStep(step_ref);
+      }
+
+      // add information about current tool
+      // register a score type
+      IdentificationData::ScoreType mass_error_ppm_score("MassErrorPPMScore", false);
+      mass_error_ppm_score_ref = id.registerScoreType(mass_error_ppm_score);
+      IdentificationData::ScoreType mass_error_Da_score("MassErrorDaScore", false);
+      mass_error_Da_score_ref = id.registerScoreType(mass_error_Da_score);
+
+      // add the same score_refs to the ProcessingSoftware - to reference the Software with the
+      // ObservationMatch - the order is important - the most important score first.
+      std::vector<IdentificationDataInternal::ScoreTypeRef> assigned_scores{mass_error_ppm_score_ref, mass_error_Da_score_ref};
+
+      // register software (connected to score)
+      // CVTerm will be set in mztab-m based on the name
+      // if the name is not available in PSI-OBO "analysis software" will be used.
+      IdentificationData::ProcessingSoftware sw("AccurateMassSearch", VersionInfo::getVersion(), assigned_scores);
+      IdentificationData::ProcessingSoftwareRef sw_ref = id.registerProcessingSoftware(sw);
+
+      // all supported search settings
+      IdentificationData::DBSearchParam search_param;
+      search_param.database = database_name_;
+      search_param.database_version = database_version_;
+      search_param.setMetaValue("database_location", database_location_);
+
+      search_param.precursor_mass_tolerance = this->mass_error_value_;
+      search_param.precursor_tolerance_ppm = this->mass_error_unit_ == "ppm" ? true : false;
+      IdentificationData::SearchParamRef search_param_ref = id.registerDBSearchParam(search_param);
+
+      // file has been processed by software performing a specific processing action.      
+      std::set<DataProcessing::ProcessingAction> actions;
+      actions.insert(DataProcessing::IDENTIFICATION);
+      IdentificationData::ProcessingStep step(sw_ref, file_refs, DateTime::now(), actions);
+      step_ref = id.registerProcessingStep(step, search_param_ref);
+      id.setCurrentProcessingStep(step_ref); // add the new step
+    }
+
```

Comment:
Indentation seems off below this point (lines 665-678).

Reply:
When I view the complete file (here or in my editor) it seems to be correct. I think it is an artifact of the representation in review mode.